free-4.md

```diff
@@ -0,0 +1,82 @@
+the user?_
+
+When the program in my language runs, it will read in the error message from the users program, pick out the relevant information for the user and print it out in full sentences for the user alonge with suggestions for what might be causing the error. An error could occur of my language does not recognize the error message output by the program, but in general my language would do all its work in the error domain.
+
+### Expressiveness
+_What should be easy to do in this language? What should be possible, but
+difficult? What should be impossible or very difficult?_
+
+It should be easy to get more information about the error, where it occured, likely reasons for its occurance, and clear, full sentences in English explaining the problem like a human tutor might. It should be impossibly to write programs in it that don't involve error checking.
+
+### Related work
+_Are there any other DSLs in this domain? If not, describe how you know there
+aren't and conjecture why not. If so, describe them and provide links. How well 
+do they address the need? Are there any particularly admirable qualities of the
+language? Are there parts of the language you think could be improved?_
+
+There do not seem to be any DSLs which quite do this. There are many debuggers like valgrind and gdb listed [here](https://en.wikipedia.org/wiki/Debugger) but they seemed to be geared towards very "technical" people, who are creating huge code bases that are very difficult to debug and therefore need the detail these dubuggers provide. None of these seem to focus on easing new coders into debugging, perhaps because most new coders learn in a setting like college where there are tutors who can help them or they just google the error. These tools were therefore mostly created by programmers working for big technical companies to be used on the sorts of projects they use and less to be used on "simply" things.
+
+## The Project
+This section examines whether the idea makes for a good CS 111 project.
+
+
+### Suitability
+_If someone were to work on this project, what percentage of their time would be
+spent directly engaging in the **language** aspects of this project (e.g.,
+making language design decisions), as opposed to "systems" aspects of the
+project (e.g., implementing a complicated semantics that doesn't require a lot
+of language design)?_
+
+This language would have very few things the user would need to write and most of the language writers time would be spent working on the back end to ensure that error messages were being correctly reinterpretted for the user.
+
+### Scope
+_How big an idea is this? How ambitious is this project?_
+
+This seems like a very big idea, but also a very intersting one. Tools like valgrind and gdb seem useful, but like the learning curve is extremely high. I would like to lower that learning curve but I would guess that it would take much longer than we have to do it well.
+
```

Comment:
Maybe instead of trying to make an explainer for all of gdb or valgrind, you could focus on one specific type of error? This might help narrow the focus and make the project less overwhelming.